ash-window/src/lib.rs

```diff
@@ -21,54 +22,55 @@ use raw_window_handle::{HasRawWindowHandle, RawWindowHandle};
 pub unsafe fn create_surface(
     entry: &Entry,
     instance: &Instance,
-    window_handle: &dyn HasRawWindowHandle,
+    display_handle: RawDisplayHandle,
+    window_handle: RawWindowHandle,
```

Comment:
Do we need to add a safety precondition that these handles are live and associated with eachother?

Reply:
I haven't checked if the API even allows for this.

Reply:
I'm not sure I follow. This is part of ash-window's public API.

Reply:
I thought you were requesting to programmatically uphold said constraints, but I think this is a request to simply document it under `# Safety` in the docs no?

Reply:
Wayland requires the surface must belong to the display. So yes the two must be associated.

Reply:
> Wayland requires the surface must belong to the display. So yes the two must be associated.

No way!

Reply:
@Ralith English-fu has never been my strong suit, but turned this into:

```rust
/// `window_handle` must be associated with `display_handle` for this call to be valid (e.g. the
/// given `RawWindowHandle` must have been created from the display connection described by
/// `RawDisplayHandle`).
```

Reply:
I'd cut that down to just "`window_handle` must be associated with `display_handle`" for concision, but that seems basically fine. Still need a note that the handle must be currently valid (e.g. not associated with a destroyed window or display connection).

Reply:
```rust
/// The window represented by `window_handle` must be associated with the display connection
/// in `display_handle`.
///
/// `window_handle` and `display_handle` must be associated with a valid window and display
/// connection, that have both not been destroyed.
```

Reply:
We should probably clarify that the window and display handles don't just need to be valid and not destroyed at creation time, but actually valid for the entire lifetime of the returned Surface.

Reply:
Sure, but it's already at a point where one can write a 1500 page manual on this. For the most part `ash`'s stance is that you're familiar with Vulkan and all its rules/caveats anyway - same applies for the window<->surface<->swapchain part IMO.